src/test/scala/io/iohk/ethereum/jsonrpc/DebugServiceSpec.scala

```diff
@@ -3,24 +3,31 @@ package io.iohk.ethereum.jsonrpc
 import java.net.InetSocketAddress
 
 import akka.actor.ActorSystem
-import akka.testkit.TestProbe
-import io.iohk.ethereum.Fixtures
+import akka.testkit.{TestKit, TestProbe}
+import io.iohk.ethereum.{Fixtures, WithActorSystemShutDown}
 import io.iohk.ethereum.jsonrpc.DebugService.{ListPeersInfoRequest, ListPeersInfoResponse}
 import io.iohk.ethereum.network.EtcPeerManagerActor.PeerInfo
 import io.iohk.ethereum.network.PeerManagerActor.Peers
 import io.iohk.ethereum.network.p2p.messages.CommonMessages.Status
 import io.iohk.ethereum.network.p2p.messages.Versions
 import io.iohk.ethereum.network.{EtcPeerManagerActor, Peer, PeerActor, PeerManagerActor}
+import monix.execution.Scheduler.Implicits.global
 import org.scalamock.scalatest.MockFactory
 import org.scalatest.concurrent.ScalaFutures
-import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.flatspec.AnyFlatSpecLike
 import org.scalatest.matchers.should.Matchers
 
-class DebugServiceSpec extends AnyFlatSpec with Matchers with MockFactory with ScalaFutures {
+class DebugServiceSpec
+    extends TestKit(ActorSystem("ActorSystem_DebugServiceSpec"))
+    with AnyFlatSpecLike
+    with WithActorSystemShutDown
+    with Matchers
+    with MockFactory
+    with ScalaFutures {
 
   "DebugService" should "return list of peers info" in new TestSetup {
-    val result: ServiceResponse[ListPeersInfoResponse] =
-      debugService.listPeersInfo(ListPeersInfoRequest())
+    val result =
+      debugService.listPeersInfo(ListPeersInfoRequest()).runToFuture
```

Comment:
Why `runToFuture`? To be consistent we should use `runSyncUnsafe` everywhere (and avoid `futureValue`)

Reply:
runSyncUnsafe  was implemented at the beginning. But this break the tests:
```
[info] DebugServiceSpec:
[info] DebugService
[info] - should return list of peers info *** FAILED *** (8 seconds, 376 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsg while waiting for PeerInfoRequest(PeerId(testProbe-4))
[info]   at scala.Predef$.assert(Predef.scala:223)
[info]   at akka.testkit.TestKitBase.expectMsg_internal(TestKit.scala:459)
[info]   at akka.testkit.TestKitBase.expectMsg(TestKit.scala:436)
[info]   at akka.testkit.TestKitBase.expectMsg$(TestKit.scala:436)
[info]   at akka.testkit.TestKit.expectMsg(TestKit.scala:969)
[info]   at io.iohk.ethereum.jsonrpc.DebugServiceSpec$$anon$1.<init>(DebugServiceSpec.scala:35)
[info]   at io.iohk.ethereum.jsonrpc.DebugServiceSpec.$anonfun$new$1(DebugServiceSpec.scala:28)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   ...
```
Some tests contain a mix of:
* expectations on Akka TestProbe
* sending messages to actors
* assertions at the end.

In some cases, I moved expectations and assertions at the end. But in some cases, I was not able to do it so I used conversion to Future to get async behavior.